docs/DeveloperGuide.md

```diff
@@ -2,37 +2,381 @@
+- If any parameters are missing (e.g., no /cal provided), they are auto-filled using the current values stored 
+in the Goals singleton instance.
+- ParserParameters extracts the parameters into a key-value map.
+- AddGoalCommand is then created with the parsed values and executed.
+- If the values differ from the current ones, Goals.updateGoals() updates the stored values, and the goal is 
+logged to the goalLogs list.
+- Track function (called with track goal /d <date>) as well as the view function (called with view goal) are 
+supplementary functions provided to increase the accountability of the user by giving him easy access to previous and
+current stats.
+
+Exceptions such as InvalidGoalException and InvalidDateFormatException are handled gracefully to ensure robustness.
+
+### 3. Why This Design
+- Simplicity: A single command (add goal) handles all updates in one go, keeping the interaction concise 
+and easy to use.
+- Singleton Pattern: Using Goals.getInstance() ensures centralized access to the current user goals across the app.
+- Auto-Fill Support: Missing parameters default to current goal values, streamlining repeated updates.
+- Separation of Concerns: Goal parsing and updating are kept modular and separate from logging and UI logic
+
+### 4. Alternatives Considered
+- Command Pattern: We considered creating individual commands for each goal type (e.g., SetWaterGoalCommand), 
+but found a unified approach more straightforward for the CLI.
+- Interactive CLI Loop: Earlier versions used a conversational loop for goal editing, but it was replaced by a 
+single-line command to reduce complexity.
+- Persistent Storage: While current goal data is stored in-memory and appended to the log file, we plan to enhance 
+persistence for full session retention.
+
+### 5. Sequence Diagrams
+![GoalSD.png](Images/GoalSD.png)
+### 6. Future Improvements
+- Input Validation: Use regex or stricter parsing to ensure only valid integers are accepted.
+- Persistent Goal Storage: Store and reload goals from file or database to retain state between sessions.
+- Multi-User Support: Refactor the singleton Goals class to support different profiles.
+- Goal Recommendations: Use history from logs (e.g., average water intake) to suggest realistic goal values
 
 ## Product scope
 ### Target user profile
 
-{Describe the target user profile}
+1. Fitness-conscious individuals
+2. Users who prefer CLI-based tools
+3. Individuals who want to log workouts, meals, water intake, and progress
+4. Java users with basic terminal skills
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+HealthBud provides a simple and effective CLI solution to track health logs including workouts, meals, water intake,
+and personal bests. Unlike complex mobile apps, it runs offline and keeps the data lightweight and easily exportable.
 
 ## User Stories
```

Comment:
The user stories could be more diverse in terms of profiles (ie make them more specific to different types of users), other than that they cover all functionalities